web/src/scheduled_messages.js

```diff
@@ -98,19 +132,26 @@ export function send_request_to_schedule_message(scheduled_message_data, deliver
     });
 }
 
-export function delete_scheduled_message(scheduled_msg_id, edit_if_delete_successs) {
+export function edit_scheduled_message(scheduled_message_id) {
+    const scheduled_msg = scheduled_messages_data.find(
+        (msg) => msg.scheduled_message_id === scheduled_message_id,
+    );
+    delete_scheduled_message(scheduled_message_id, () =>
+        open_scheduled_message_in_compose(scheduled_msg),
+    );
+}
+
+export function delete_scheduled_message(scheduled_msg_id, success = () => {}) {
     channel.del({
         url: "/json/scheduled_messages/" + scheduled_msg_id,
-        success() {
-            // TODO: Do this via events received from the server in server_events_dispatch.
-            if (overlays.scheduled_messages_open()) {
-                $(
-                    `#scheduled_messages_overlay .scheduled-message-row[data-message-id=${scheduled_msg_id}]`,
-                ).remove();
-            }
-            if (edit_if_delete_successs) {
-                edit_scheduled_message(scheduled_msg_id);
-            }
-        },
+        success,
     });
 }
+
+export function initialize() {
+    if (scheduled_messages_data.length === 0) {
+        scheduled_messages_data = page_params.scheduled_messages;
+    } else {
+        add_scheduled_messages(page_params.scheduled_messages);
+    }
```

Comment:
Under what circumstances will this `if/else` be relevant? This function is only called once during startup, right -- is it for tests to reset the state? A comment would be helpful if so.

Reply:
This is the handle the case of client receiving an event before we can initialize `scheduled_messages_data`. Added that comment in #25390.